### PR TITLE
feat(Editor): Add deprecated support

### PR DESCRIPTION
**Pull request by aoudiamoncef**
Hi @EricWittmann 

Please don't laugh, i' definitely not a front-end developer. Don't hesitate to give me some feedbacks about this feature. 

Screenshot from 2020-04-05 19-28-15
Screenshot from 2020-04-05 19-28-38
Screenshot from 2020-04-05 19-29-02

### CODE REVIEW

File: front-end/studio/src/app/pages/apis/{apiId}/editor/_components/forms/path/operation/info-section.component.html [hunk 1]
```diff
@@ -32,6 +32,14 @@
                                        [baseNode]="operation" nodePath="operationId"
                                        (onChange)="changeOperationId($event)"></pf-inline-text-editor>
             </div>
+
+            <!-- Deprecated -->
+            <div class="section-field deprecated">
+                <drop-down [value]="operation.deprecated"
+                           [options]="deprecatedOptions()"
+                           (onValueChange)="changeDeprecated($event)"
+                           [noSelectionLabel]="'not-deprecated'"></drop-down>
```
Comment (EricWittmann) [inline front-end/studio/src/app/pages/apis/{apiId}/editor/_components/forms/path/operation/info-section.component.html:41]: Can you change this to:

`noSelectionLabel="Not Deprecated"`

Comment (EricWittmann) [reply front-end/studio/src/app/pages/apis/{apiId}/editor/_components/forms/path/operation/info-section.component.html:41]: In other words, you can drop the braces around `noSelectionLabel` and then just have the value in quotes (without the single quotes).  And then the value should match what you have in the dropdown options later on. :)

File: front-end/studio/src/app/pages/apis/{apiId}/editor/_components/forms/path/operation/info-section.component.ts [hunk 2]
```diff
@@ -148,6 +149,16 @@ export class OperationInfoSectionComponent extends AbstractBaseComponent {
...
     }
 
+    /**
+     * Called when the user changes the deprecated.
+     * @param newDeprecated
+     */
+    public changeDeprecated(newDeprecated: boolean): void {
+        console.info("[InfoSectionComponent] User changed the deprecated.");
+        let command: ICommand = CommandFactory.createChangePropertyCommand(this.operation, "deprecated", newDeprecated);
+        this.commandService.emit(command);
```
Comment (EricWittmann) [inline front-end/studio/src/app/pages/apis/{apiId}/editor/_components/forms/path/operation/info-section.component.ts:159]: Nice.

File: front-end/studio/src/app/pages/apis/{apiId}/editor/_services/rest-resource.service.ts [hunk 3]
```diff
@@ -71,6 +71,7 @@ export class RestResourceService {
         // The GET operation
         let baseGetOp: OasOperation = basePathItem.createOperation("get");
         basePathItem.get = baseGetOp;
+        baseGetOp.deprecated = false;
```
Comment (EricWittmann) [inline front-end/studio/src/app/pages/apis/{apiId}/editor/_services/rest-resource.service.ts:74]: Do you think we need to add these?  The default value for `deprecated` is `false` anyway.

**Comment by EricWittmann** [comment]
Hey @aoudiamoncef the maven build is failing for this.  If you get a chance can you have a look?  If not I'll get to it when I can.  FYI the build is less permissive than the UI dev server, so sometimes the build will fail even if everything seems to work at dev time. :)

**Comment by aoudiamoncef** [comment]
Hi @EricWittmann,
I rebased from the current failing commit in master, I don't think that the PR modification could break the build.

**Comment by EricWittmann** [comment]
OK I'll try it out locally.

**Comment by EricWittmann** [comment]
OK I found the problem.  You had removed the single-quotes from that one attribute in the template, but kept the square braces.  No worries - I've fixed it and will merge soon.

**Comment by EricWittmann** [comment]
All done, thanks for the contribution @aoudiamoncef

**Pull request merged**